src/ray/common/asio/instrumented_io_context.h

```diff
@@ -61,6 +61,23 @@ struct GuardedGlobalStats {
   mutable absl::Mutex mutex;
 };
 
+/// An opaque stats handle, used to manually instrument event loop handlers that don't
+/// hook into a .post() call.
+struct StatsHandle {
+  std::string handler_name;
+  int64_t start_time;
+  std::shared_ptr<GuardedHandlerStats> handler_stats;
+  std::shared_ptr<GuardedGlobalStats> global_stats;
+
+  StatsHandle(std::string handler_name_, int64_t start_time_,
+              std::shared_ptr<GuardedHandlerStats> handler_stats_,
```

Comment:
```suggestion
              std::shared_ptr<GuardedHandlerStats> &handler_stats_,
```
They will be copied when 

```
        handler_stats(handler_stats_),
        global_stats(global_stats_) {}
```
happens?

Reply:
So the convention here is to take the shared pointer by copy in the constructor argument, and `std::move()` them into the member field. I forgot to `std::move()` these, so I'll make that update!